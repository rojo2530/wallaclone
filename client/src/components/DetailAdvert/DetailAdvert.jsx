import React from 'react';
import Loading from '../Loading/';
import Navbar from '../Navbar/';
import Footer from '../Footer/';
import { FaCoins, FaShoppingCart, FaTruck } from 'react-icons/fa';
import CaptureError  from '../CaptureError/';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';

class DetailAdvert extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const { id } = this.props.match.params;
    this.props.loadAdvert(id);
  }

  render() {
    const { advert, isFetching, error, t }  = this.props;
    if (isFetching) {
      return <Loading text='Fetching detail Advert' /> 
    }
    if (error) {
      return <CaptureError message="Error fecthing Advert" error={error.message} />
    }
    if (!advert) {
      return null;
    }
    return (
      <>
      <Navbar />
        <section className="section detail-container has-spacing section-gray">
          <div className="container">
            <div className="columns">
              <div className="column">
                <div className="image is-5by4">
                  <img src={advert.photo.startsWith('/images') ? `http://localhost:3001${advert.photo}` : `${advert.photo}`} alt="Placeholder" />
                </div>
              </div>
              <div className="column">
                <div className="card">
                  <div className="card-content">
                    <h1 className="title">{advert.name}</h1>
                    <div className="content">
                      <p className="is-size-5">â‚¬{advert.price}</p>
                      <h6 className="vc">{advert.type}</h6>
                      <div className="has-spacing-bottom">
                      {advert.tags.map(tag => (
                          <span key={tag} className="tag has-small-spacing-top is-medium">{tag}</span>
                        ))}
                      </div>
    
                      <p>{advert.description}</p>                            <p className="buttons">
                        <a className="button is-link has-icons-left is-medium" href="/">
                          <span className="icon">
                            <FaShoppingCart />
                          </span>
                          <span>Add to basket</span>
                        </a>
                      </p>
                      <hr />
                      <div className="media">
                        <div className="media-left">
                          <span className="icon">
                            <FaCoins />
                          </span>
    
                        </div>
                        <div className="media-content">
                          <p className="title is-5">{t("Money Back Guarantee")}</p>
                          <p className="subtitle is-5">Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae</p>
                        </div>
                      </div>
                      <hr />
                      <div className="media">
                        <div className="media-left">
                          <span className="icon">
                            <FaTruck />
                          </span>
    
                        </div>
                        <div className="media-content">
                          <p className="title is-5">{t("International Delivery")}</p>
                          <p className="subtitle is-5">Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae</p>
                        </div>
                      </div>
    
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <Footer />
      </>
    )
  }
}

DetailAdvert.propTypes = {
  advert: PropTypes.object,
  error: PropTypes.object,
  isFetching: PropTypes.bool,
  loadAdvert: PropTypes.func.isRequired,
}

export default withTranslation()(DetailAdvert);
