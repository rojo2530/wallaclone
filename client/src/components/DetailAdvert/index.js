import DetailAdvert from './DetailAdvert';
import { connect } from 'react-redux';
import { fecthSingleAdvert } from '../../store/actions';

function mapStateToProps(state) {
  return {
    advert: state.currentAdvert,
    isFetching: state.ui.isFetching,
    error: state.ui.error,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    loadAdvert: id => dispatch(fecthSingleAdvert(id)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DetailAdvert)