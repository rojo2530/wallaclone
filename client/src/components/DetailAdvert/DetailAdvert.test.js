import React from 'react';
import { shallow } from 'enzyme';
import DetailAdvert from './DetailAdvert';
import { FaSellcast, FaBuysellads, FaItalic } from 'react-icons/fa';

describe('DetailAdvert', () => {
  const  defaultProps = {
    loadAdvert: jest.fn(),
    isFetching: false,
    error: null,
    match: {
      params: 1
    },
    advert: {
      photo: '/images/prueba.png',
      name: 'prueba',
      price: 15,
      type: 'buy' ,
      tags: ['work']
    }
  }

  const render = props => shallow(<DetailAdvert {...defaultProps} {...props} />);
  let wrapper;

  beforeEach(() => {
    wrapper = render();
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should call to prop loadAdvert on ComponentDidMount', () => {
    const instance = wrapper.instance();
    const { loadAdvert } = defaultProps;
    const { match } = defaultProps; 
    instance.componentDidMount();
    expect(loadAdvert).toHaveBeenCalledWith(match.id);
  });
});