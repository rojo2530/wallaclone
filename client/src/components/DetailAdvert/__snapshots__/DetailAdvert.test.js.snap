// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailAdvert should render 1`] = `
<Fragment>
  <Connect(withRouter(Navbar)) />
  <section
    className="section detail-container has-spacing section-gray"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <div
            className="image is-5by4"
          >
            <img
              alt="Placeholder"
              src="http://localhost:7000/images/prueba.png"
            />
          </div>
        </div>
        <div
          className="column"
        >
          <div
            className="card"
          >
            <div
              className="card-content"
            >
              <h1
                className="title"
              >
                prueba
              </h1>
              <div
                className="content"
              >
                <p
                  className="is-size-5"
                >
                  â‚¬
                  15
                </p>
                <h6
                  className="vc"
                >
                  buy
                </h6>
                <div
                  className="has-spacing-bottom"
                >
                  <span
                    className="tag has-small-spacing-top is-medium"
                    key="work"
                  >
                    work
                  </span>
                </div>
                <p />
                                            
                <p
                  className="buttons"
                >
                  <a
                    className="button is-link has-icons-left is-medium"
                    href="/"
                  >
                    <span
                      className="icon"
                    >
                      <FaShoppingCart />
                    </span>
                    <span>
                      Add to basket
                    </span>
                  </a>
                </p>
                <hr />
                <div
                  className="media"
                >
                  <div
                    className="media-left"
                  >
                    <span
                      className="icon"
                    >
                      <FaCoins />
                    </span>
                  </div>
                  <div
                    className="media-content"
                  >
                    <p
                      className="title is-5"
                    >
                       Money Back Guarantee
                    </p>
                    <p
                      className="subtitle is-5"
                    >
                      Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae
                    </p>
                  </div>
                </div>
                <hr />
                <div
                  className="media"
                >
                  <div
                    className="media-left"
                  >
                    <span
                      className="icon"
                    >
                      <FaTruck />
                    </span>
                  </div>
                  <div
                    className="media-content"
                  >
                    <p
                      className="title is-5"
                    >
                       International Delivery
                    </p>
                    <p
                      className="subtitle is-5"
                    >
                      Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <Footer />
</Fragment>
`;
